cmd/goldpinger/main.go

```diff
@@ -96,8 +96,17 @@ func main() {
 		goldpinger.GoldpingerConfig.Port = server.Port
 	}
 
+	if GoldpingerConfig.PodIP == "" {
+		log.Println("PodIP not set: pinging all pods")
+	}
+	if GoldpingerConfig.PingNumber == 0 {
+		log.Println("--ping-number set to 0: pinging all pods")
+	}
+	podSelecter := newPodSelecter(GoldpingerConfig.PingNumber, GoldpingerConfig.PodIP, GetAllPods)
+
+	// TODO: How can we inject podselecter here?
 	server.ConfigureAPI()
```

Comment:
I'm having trouble figuring out how to deal with the swagger generated code. The only file dealing with server stuff that I seem to be able to edit is `configure_goldpinger.go`, but the function call I want to update `configureAPI()` is being used by a generated file that I shouldn't update.

Any guidance on how to handle this?

Reply:
That's correct - the place to look at is [`configure_goldpinger.go`](https://github.com/bloomberg/goldpinger/blob/master/pkg/restapi/configure_goldpinger.go#L70) and the `ConfigureAPI` should be safe to edit, as I can see you've already figured out :-)

Reply:
`configureAPI` ended up being relatively inflexible because it's called by a method in an autogenerated file that it sounds like I shouldn't edit. I made some changes in `configure_goldpinger.go` that we can discuss, potentially there's a way to introduce my changes that doesn't involve more or less copying one of the methods.

Reply:
`configure_goldpinger.go`, once exists, doesn't get overwritten. So as long as you don't change its signature, it should be safe to edit `configureAPI`. It is called from the other generated files, but that's fine.

I __much__ prefer the previous approach, which doesn't involve copying the entire function.

Reply:
If you're just uncomfortable passing the `podSelecter` instance around, you can just stick it in `https://github.com/bloomberg/goldpinger/blob/master/pkg/goldpinger/config.go#L30` the config, just like we do with the `kubernetes` client.

Reply:
no worries, updated